velox/exec/HashProbe.cpp

```diff
@@ -320,10 +320,13 @@ void HashProbe::asyncWaitForHashTable() {
   checkRunning();
   VELOX_CHECK_NULL(table_);
 
+  // Release any reserved memory before wait for next round of hash join in case
+  // of disk spilling has been triggered.
+  pool()->release();
+
   auto hashBuildResult = joinBridge_->tableOrFuture(&future_);
   if (!hashBuildResult.has_value()) {
     VELOX_CHECK(future_.valid());
-    pool()->release();
```

Comment:
So if current processing partition of build side has spilled, this pool will never be released unless another reclaim triggers the release in ::reclaim() ?
Why don't we release the pool after we done processing the output?

Reply:
This change just unconditionally release memory